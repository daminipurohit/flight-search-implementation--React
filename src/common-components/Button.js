import React from 'react';
import PropTypes from 'prop-types';
import '../styles/button.scss';

const getClassName = ({ className, primary, danger, disabled }) => {
    let btnClass = `button-default ${className}`;
    if (primary) {
        btnClass += ' button-primary';
    }
    if (danger) {
        btnClass += ' button-danger';
    }
    if (disabled) {
        btnClass += ' button-disabled';
    }
    return btnClass;
};

const Button = ({ className = '', label, onClick, primary, danger, disabled = false }) => {
    const buttonClass = getClassName({ className, primary, danger, disabled });
    return (
        <button className={buttonClass} onClick={onClick} disabled={disabled}>{label}</button>
    );
};

Button.propTypes = {
    label: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    primary: PropTypes.bool,
    danger: PropTypes.bool,
    className: PropTypes.string,
    disabled: PropTypes.bool
};

export default Button;
