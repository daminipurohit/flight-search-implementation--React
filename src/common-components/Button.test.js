import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
    it('should render correctly as default button', () => {
        const onClick = jest.fn();
        const props = {
            label: 'Default button',
            onClick
        };
        const component = shallow(<Button {...props} />);
        expect(component).toMatchSnapshot();
    });

    it('should render correctly as primary button', () => {
        const onClick = jest.fn();
        const props = {
            label: 'Primary button',
            onClick,
            primary: true
        };
        const component = shallow(<Button {...props} />);
        component.find('button').simulate('click');
        expect(onClick).toHaveBeenCalled();
        expect(component).toMatchSnapshot();
    });

    it('should render correctly as danger button', () => {
        const onClick = jest.fn();
        const props = {
            label: 'Danger button',
            onClick,
            danger: true
        };
        const component = shallow(<Button {...props} />);
        component.find('button').simulate('click');
        expect(onClick).toHaveBeenCalled();
        expect(component).toMatchSnapshot();
    });

    it('should render correctly as disabled button', () => {
        const onClick = jest.fn();
        const props = {
            label: 'Disabled button',
            onClick,
            disabled: true
        };
        const component = shallow(<Button {...props} />);
        expect(component.find('button').prop('disabled')).toBe(true);
        expect(component.find('button').hasClass('button-default  button-disabled')).toBe(true);
        expect(component).toMatchSnapshot();
    });
});
