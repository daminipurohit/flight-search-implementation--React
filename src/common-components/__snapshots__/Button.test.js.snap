// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly as danger button 1`] = `
<button
  className="button-default  button-danger"
  disabled={false}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  Danger button
</button>
`;

exports[`Button should render correctly as default button 1`] = `
<button
  className="button-default "
  disabled={false}
  onClick={[MockFunction]}
>
  Default button
</button>
`;

exports[`Button should render correctly as disabled button 1`] = `
<button
  className="button-default  button-disabled"
  disabled={true}
  onClick={[MockFunction]}
>
  Disabled button
</button>
`;

exports[`Button should render correctly as primary button 1`] = `
<button
  className="button-default  button-primary"
  disabled={false}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  Primary button
</button>
`;
